### PR TITLE
Adding ubuntu c3os image changes, also overlay files for different pr…

### PR DESCRIPTION
**Pull request by vipsharm**
Ubuntu OS support and different provider overlay files.

### CODE REVIEW

File: images/Dockerfile.ubuntu [hunk 1]
```diff
@@ -10,41 +10,45 @@ RUN apt install -y software-properties-common
...
+    debianutils \
+    curl \
+    openssh-server \
+    nano \
+    gawk \
+    haveged \
+    rsync \
+    neovim \
+    open-vm-tools \
+    netplan.io \
```
Comment (mudler) [inline images/Dockerfile.ubuntu:37]: any specific reason for netplan/wireguard?

File: overlay/k8s/cncf/etc/k8sconfig/calico.yaml [hunk 2]
```diff
@@ -0,0 +1,4648 @@
+---
+# Source: calico/templates/calico-config.yaml
+# This ConfigMap is used to configure a self-hosted Calico installation.
+kind: ConfigMap
```
Comment (mudler) [inline overlay/k8s/cncf/etc/k8sconfig/calico.yaml:4]: this doesn't belong here - in the core there shouldn't be any provider specific configuration

File: Earthfile [hunk 3]
```diff
@@ -226,6 +227,9 @@ docker:
         COPY overlay/files-opensuse-arm-rpi/ /
     ELSE IF [ "$FLAVOR" = "opensuse-arm-rpi" ]
         COPY overlay/files-opensuse-arm-rpi/ /
+    ELSE IF [ "$FLAVOR" = "ubuntu" ]
+        COPY overlay/files-ubuntu/ /
+        COPY overlay/k8s/$K8S_FLAVOR/ /
```
Comment (mudler) [inline Earthfile:232]: same, the overlay files here are very specific to the kubernetes engines - those changes are not meant to be here

**Comment by mudler** [comment]
Also please enable ubuntu into the pipelines so we can see if they pass tests

File: overlay/k8s/cncf/etc/k8sconfig/calico.yaml (hunk 2 above)
Comment (vipsharm) [reply overlay/k8s/cncf/etc/k8sconfig/calico.yaml:4]: Ok, so better to move provider overlay to provider repo ? I will fix this

**Comment by mudler** [comment]
the whole `overlay/k8s` folder is really specific and seems should go elsewhere, otherwise seems to go into the right direction

File: overlay/k8s/cncf/etc/k8sconfig/calico.yaml (hunk 2 above)
Comment (mudler) [reply overlay/k8s/cncf/etc/k8sconfig/calico.yaml:4]: yes exactly, this is provider specific

File: Earthfile (hunk 3 above)
Comment (vipsharm) [reply Earthfile:232]: I will remove these.

File: images/Dockerfile.ubuntu (hunk 1 above)
Comment (vipsharm) [reply images/Dockerfile.ubuntu:37]: I used the OS file from PaletteOS that Venkat added for some earlier POC I believe. I will check with him if we need these.

Comment (mudler) [reply images/Dockerfile.ubuntu:37]: those can be added later on when creating the image for the provider, no need to add them here

Comment (mudler) [inline images/Dockerfile.ubuntu:37]:
```suggestion

```

File: images/Dockerfile.ubuntu [hunk 4]
```diff
@@ -10,41 +10,45 @@ RUN apt install -y software-properties-common
...
+    wireguard \
```
Comment (mudler) [inline images/Dockerfile.ubuntu:38]:
```suggestion
```

**Comment by mudler** [comment]
please rebase/squash commits

File: images/Dockerfile.ubuntu [hunk 5]
```diff
@@ -33,9 +33,6 @@ RUN apt install -y \
     haveged \
     rsync \
     neovim \
-    open-vm-tools \
```
Comment (mudler) [inline images/Dockerfile.ubuntu]: can keep `open-vm-tools`, this one is for vmware guest integration

Comment (mudler) [reply images/Dockerfile.ubuntu]: but either looks fine to me

**Review by mudler — APPROVED** [review]

**Comment by mudler** [comment]
Looking good, thanks!

**Pull request merged**